packages/dev/serializers/src/glTF/2.0/Extensions/EXT_mesh_gpu_instancing.ts

```diff
@@ -0,0 +1,180 @@
+                            AccessorComponentType.FLOAT
+                        );
+                    }
+                    // do we need to write ROTATION ?
+                    if (hasAnyInstanceWorldRotation) {
+                        // Data type can be
+                        //   - 5126 (FLOAT)
+                        //   - 5120 (BYTE) normalized
+                        //   - 5122 (SHORT) normalized
+                        // this is defined by option.
+                        let componentType = this._exporter.options?.meshGpuInstancingOptions?.quaternionType ?? AccessorComponentType.FLOAT;
+                        if (componentType != AccessorComponentType.FLOAT && componentType != AccessorComponentType.SHORT && componentType != AccessorComponentType.BYTE) {
+                            // force to float if wrong type.
+                            componentType = AccessorComponentType.FLOAT;
+                        }
+                        gpu_instancing.attributes["ROTATION"] = this._buildAccessor(rotationBuffer, AccessorType.VEC4, babylonNode.thinInstanceCount, binaryWriter, componentType);
+                    }
+                    // do we need to write SCALE ?
+                    if (hasAnyInstanceWorldScale) {
+                        gpu_instancing.attributes["SCALE"] = this._buildAccessor(
+                            scaleBuffer,
+                            AccessorType.VEC3,
+                            babylonNode.thinInstanceCount,
+                            binaryWriter,
+                            AccessorComponentType.FLOAT
+                        );
+                    }
+
+                    /* eslint-enable @typescript-eslint/naming-convention*/
+                    node.extensions = node.extensions || {};
+                    node.extensions[NAME] = gpu_instancing;
+                }
+            }
+            resolve(node);
+        });
+    }
+
+    private _buildAccessor(
+        buffer: Float32Array | Int8Array | Int16Array,
+        type: AccessorType,
+        count: number,
+        binaryWriter: _BinaryWriter,
+        componentType: AccessorComponentType
+    ): number {
+        // write the buffer
+        const bufferOffset = binaryWriter.getByteOffset();
+        switch (componentType) {
+            case AccessorComponentType.FLOAT: {
+                for (let i = 0; i != buffer.length; i++) {
+                    binaryWriter.setFloat32(buffer[i]);
+                }
+                break;
+            }
+            case AccessorComponentType.BYTE: {
+                for (let i = 0; i != buffer.length; i++) {
+                    binaryWriter.setByte(buffer[i] * 127);
+                }
+                break;
+            }
+            case AccessorComponentType.SHORT: {
+                for (let i = 0; i != buffer.length; i++) {
+                    binaryWriter.setInt16(buffer[i] * 32767);
+                }
+                break;
+            }

```

Comment:
Does this work? I'm confused. If the original buffer is a normalized Int8Array or Int16Array, don't you have to divide by 127 and 32767?

Reply:
we suppose to have always double (number) value as input, which may be store at int value in case of normalized input. 
So we might multiply by 127 or 32767, depending BYTE or SHORT target.
The fact that the number is representing a normalized value or not is by the gltf component type of the accessor (by [definition ](https://github.com/KhronosGroup/glTF/blob/main/extensions/2.0/Vendor/EXT_mesh_gpu_instancing/README.md)of the extension)

Reply:
Oh, I missed that the output is byte/short, but still, this is confusing. Are the inputs always float? If so, then why `buffer: Float32Array | Int8Array | Int16Array`?

Reply:
Yes, you're right, i missed this point. I introduce it first because i believed that quaternion should be normalized into Babylon, but right after i realized that it will be never the case
we manipulate 

- TranslationBuffer which is always float
- ScaleBuffer which is alway float
- and RotationBuffer which is always float

BECAUSE it's all comming from matrix.decompose.

So i may change the 
`buffer: Float32Array | Int8Array | Int16Array` to `buffer: Float32Array`